lib/spack/spack/cmd/env.py

```diff
@@ -380,28 +380,35 @@ def env_remove(args):
     and manifests embedded in repositories should be removed manually.
     """
     read_envs = []
+    bad_envs = []
     for env_name in args.rm_env:
-        env = ev.read(env_name)
-        read_envs.append(env)
-
-    if not args.yes_to_all:
-        answer = tty.get_yes_or_no(
-            "Really remove %s %s?"
-            % (
-                string.plural(len(args.rm_env), "environment", show_n=False),
-                string.comma_and(args.rm_env),
-            ),
-            default=False,
-        )
-        if not answer:
-            tty.die("Will not remove any environments")
-
-    for env in read_envs:
-        if env.active:
-            tty.die("Environment %s can't be removed while activated." % env.name)
-
-        env.destroy()
-        tty.msg("Successfully removed environment '%s'" % env.name)
+        try:
+            env = ev.read(env_name)
+            read_envs.append(env)
+        except spack.config.ConfigFormatError:
+            bad_envs.append(env_name)
+
+        if not args.yes_to_all:
+            environments = string.plural(len(args.rm_env), "environment", show_n=False)
+            envs = string.comma_and(args.rm_env)
+            answer = tty.get_yes_or_no(f"Really remove  {environments} {envs}?", default=False)
+            if not answer:
+                tty.die("Will not remove any environments")
+
+        for env in read_envs:
+            name = env.name
+            if env.active:
+                tty.die(f"Environment {name} can't be removed while activated.")
+            env.destroy()
+            tty.msg(f"Successfully removed environment {name}")
+
+        for bad_env_name in bad_envs:
+            shutil.rmtree(
+                spack.environment.environment.environment_dir_from_name(
+                    bad_env_name, exists_ok=True
+                )
+            )
+            tty.msg(f"Successfully removed environment '{bad_env_name}'")
```

Comment:
@tldahlgren Isn't indentation one level too much?

Reply:
Indentation for which line(s)?  Let me expand my window and double check.

Reply:
The entire block that is highlighted

Reply:
The whole for-loop!  I'm going to have to expand the window more often.  

How should we handle this?

Reply:
> The entire block that is highlighted

🤦

Reply:
🤦